### DISCUSSION TITLE
trim(): Argument #1 ($string) must be of type string, array given after updating versions

### DISCUSSION THREAD

**Discussion by V3ct0rek** [filamentphp/filament#13865] [Help]
### Package

Form builder

### Package Version

3.2.43

### How can we help you?

# Odd Bug in Filament After Composer Update

## Issue Description

Yesterday, I encountered a really odd bug. After running `composer update`, the version of Filament changed from 3.2.42 to 3.2.8, and resources where we were using 

```php
Section::make('Basic data')
    ->aside()
    ->schema([]) 
```

suddenly broke. We were unable to do anything inside these sections, while resources without sections worked totally fine.

When we tried downgrading the Filament version, we had to go back to 3.2.42 to fix the issue, so something problematic occurred between versions 3.2.42 and 3.2.43.

I have no idea what happened, but in every resource with a `Section`, we were getting this error:

```
trim(): Argument #1 ($string) must be of type string, array given
```

This error occurred inside `resources/views/vendor/filament/components/section/index.blade.php` on line 45.

## Questions

Does anyone know what could have caused this? Maybe my Laravel version is incompatible, or there's another issue. 

Thanks in advance.

**Comment by leandrocfe** [comment]
Latest version is 3.2.97

**Comment by JonathanATyler** [comment]
I have seen this same issue - current version is 3.2.102.
I've tried re-publishing assets, etc.
Happens in:
- resources/views/vendor/filament/components/section/index.blade.php:45 - 53
```blade
{{
        $attributes->class([
            'fi-section',
            match ($aside) {
                true => 'fi-aside grid grid-cols-1 items-start gap-x-6 gap-y-4 md:grid-cols-3',
                false => 'rounded-xl bg-white shadow-sm ring-1 ring-gray-950/5 dark:bg-gray-900 dark:ring-white/10',
            },
        ])
    }}
```
- resources/views/vendor/filament/components/tabs/item.blade.php:54 - 65
```blade
{{
        $attributes
            ->merge([
                'aria-selected' => $active,
                'role' => 'tab',
            ])
            ->class([
                'fi-tabs-item group flex items-center gap-x-2 rounded-lg px-3 py-2 text-sm font-medium outline-none transition duration-75',
                $inactiveItemClasses => (! $hasAlpineActiveClasses) && (! $active),
                $activeItemClasses => (! $hasAlpineActiveClasses) && $active,
            ])
    }}
```
If I comment out those lines I can get the page to render. Will try digging in further and update if I find more.

**Reply by JonathanATyler** [reply]
... Hmmm, okay, so I removed my /vendor and ran composer install again, same issue. Then I removed the resources/views/vendor/filament* folder/files and the error went away. I suppose when I ran the publish filament-(forms-)-views tags it didn't overwrite them, which is expected I guess. So @V3ct0rek you may want to try that, or at least delete the problematic files (section\index... etc) and re-publish the views (if needed).

**Reply by Alezx6** [reply]
Muchas gracias. Esto me sirvió. Actualice filamentphp y me salía ese error. Hice lo que mencionaste y resulto sin problemas.

**Comment by tiagoofarias** [comment]
Just delete your \resources\views\vendor\filament directory.

**Reply by xeradi** [reply]
Its working Thanks

**Comment by KhizarJamshaidIqbal** [comment]
See i have this same issue
trim(): Argument #1 ($string) must be of type string, array given
They show that error show on when deployment on live server
https://bookingforms.epsoldev.com/booking/desert-excursion
resources/views/components/form-select.blade.php :16

COde:
"
<div class="form-group">
    @if ($label)
        <label for="{{ $name }}" class="form-label">
            {{ $label }}
            @if ($required)
                <span class="required">*</span>
            @endif
        </label>
    @endif

    <select
        name="{{ $name }}"
        id="{{ $name }}"
        class="form-select"
        {{ $required ? 'required' : '' }}
        {{ $attributes }}
    >
        @if ($placeholder)
            <option value="" disabled {{ $selected ? '' : 'selected' }}>{{ $placeholder }}</option>
        @endif

        @foreach ($options as $value => $label)
            <option value="{{ $value }}" {{ $value == $selected ? 'selected' : '' }}>{{ $label }}</option>
        @endforeach
    </select>
</div>

<style>
    .form-group {
        margin-bottom: 15px;
    }

    .form-label {
        display: block;
        margin-bottom: 5px;
        font-weight: bold;
    }

    .form-select {
        width: 100%;
        padding: 10px;
        border: 1px solid #ccc;
        border-radius: 4px;
        box-sizing: border-box;
        background-color: white;
    }

    .required {
        color: red;
    }

    .form-select:focus {
        border-color: #15BFBF;
        outline: none;
    }
</style>

"

Please guide me how to fix it?
